src/OrchardCore.Modules/OrchardCore.Navigation/PagerShapes.cs

```diff
@@ -572,5 +562,37 @@ private static IHtmlContent CoerceHtmlString(object value)
 
             return new HtmlContentString(value.ToString());
         }
+
+        private static RouteValueDictionary GetRouteData(Shape shape, DisplayContext displayContext, IHtmlHelper Html)
+        {
+            var httpContextAccessor = displayContext.ServiceProvider.GetService<IHttpContextAccessor>();
+            var httpContext = httpContextAccessor.HttpContext;
+
+            var routeData = new RouteValueDictionary(Html.ViewContext.RouteData.Values);
+
+            if (httpContext != null)
+            {
+                var queryString = httpContext.Request.Query;
+                if (queryString != null)
+                {
+                    foreach (var key in from string key in queryString.Keys where key != null && !routeData.ContainsKey(key) let value = queryString[key] select key)
+                    {
+                        routeData[key] = queryString[key];
```

Comment:
That's okay as it is from existing code.

But I would rename `queryString` to `query`, hmm here I don't think it can be null and can have any null key, then we do `let value` but don't use it. So the suggested code would be, but would need to be tried, so as you want ;)

                var query = httpContext.Request.Query;
                foreach (var key in query.Keys)
                {
                    routeData.TryAdd(key, query[key]);
                }

Reply:
Was working before ... code optimization. Idem. 😉

Reply:
![image](https://user-images.githubusercontent.com/3228637/169443866-53fab449-5efe-4980-86ec-d5d522928d85.png)

It is called a QueryString. Probably was shorter to just name it `Request.Query` but it is always called a QueryString.

Reply:
No problem for the naming

The `Request` has 2 different properties `QueryString` and `Query`, but yes the last has been populated by parsing the first.